src/providers/websql.js

```diff
@@ -133,7 +132,16 @@ WebSQL.prototype.open = function(callback) {
     return callback();
   }
 
-  var db = global.openDatabase(that.name, WSQL_VERSION, WSQL_DESC, WSQL_SIZE);
+  // Deal with size limit like pouchdb:
+  // http://pouchdb.com/2014/10/26/10-things-i-learned-from-reading-and-writing-the-pouchdb-source.html
+  // https://github.com/pouchdb/pouchdb/blob/39da6013d74685189fcdf5bc2c617db5b76e88bd/lib/adapters/websql/utils.js#L161-L175
+  var isAndroid = false;
+  if(global.navigator && global.navigator.userAgent) {
+    isAndroid = /Android/.test(window.navigator.userAgent);
+  }
+  var dbSize = isAndroid ? 5000000 : 1; // in PhantomJS, if you use 0 it will crash
```

Comment:
Read 1) at the link on line 136: "The actual cutoff to avoid the popup is 5000000 (i.e. 5 megabytes, aka 5MB), not 5*1024*1024 (5 mebibytes, aka 5MiB)."